ilxutils/ilxutils/generic_existing_ids_upload.py

```diff
@@ -0,0 +1,111 @@
+import json
+import pandas as pd
+import rdflib
+from pyontutils.utils import rdf, rdfs, owl, skos, oboInOwl, makeGraph
+from sqlalchemy import create_engine, inspect, Table, Column
+import scicrunch_client_slow_stable as scicrunch_client
+import sys
+import pandas as pd
+from args_reader import read_args
+
+args = read_args(keys='../keys.txt', production='-p')
+curr_stage = 'production'
+engine_key=args['engine_key']
+
+#sci = scicrunch_client.scicrunch(base_path='https://scicrunch.org', key='you-key')
+#engine_key = 'production-mysql-login' #
+
+def get_subj_to_tid(engine_key):
+    engine = create_engine(engine_key)
+    data =  """
+            SELECT ti.curie, t.label, ti.tid, ti.iri
+            FROM term_existing_ids AS ti
+            JOIN terms AS t
+            WHERE ti.tid=t.id
+            """
+    df = pd.read_sql(data, engine)
+    subj_to_tid = {row.iri:row.tid for row in df.itertuples()}
+    return subj_to_tid
+
+def append_existing_ids(existing_ids, tid, obj, curie):
+    iris = [existing_id['iri'] for existing_id in existing_ids]
+    if obj not in iris:
+        existing_ids += [{
+            'tid':tid,
+            'iri':obj,
+            'curie':curie,
+            'preferred':0,
+        }]
+    return existing_ids
+
+def diff_term_data(k, new_term_data, old_term_data):
+    dont_care = ['existing_ids', 'time', 'version', 'ilx', 'definition']
+
+    for key, value in new_term_data.items():
+
+        if key in dont_care: continue
+
+        if key == 'synonyms':
+            if old_term_data.get('synonyms') != None and new_term_data.get('synonyms') == None:
+                print('FAILED')
+                sys.exit(str(k)+'\n'+str(old_term_data)+'\n'+str(new_term_data))
+            else:
+                continue
+
+        if key == 'superclasses':
+            if len(old_term_data.get('superclasses')) != len(new_term_data.get('superclasses')):
+                print('FAILED')
+                sys.exit(str(k)+'\n'+str(old_term_data)+'\n'+str(new_term_data))
+            continue
+
+        if new_term_data[key] != old_term_data[key]:
+            print('FAILED')
+            print(key)
+            print(new_term_data[key])
+            print(old_term_data[key])
+            sys.exit(str(k)+'\n'+str(old_term_data)+'\n'+str(new_term_data))
+
+#subj_to_tid = get_subj_to_tid(engine_key)
+filename = 'OEN-NLXOEN'
+new_existing_ids = json.load(open('/Users/tmsincomb/Desktop/dump/Updating_Prefixes/'+filename+'.json'))
```

Comment:
Can you please find and switch all direct uses of `open` to `with open` so we don't have to worry about the number of dangling file descriptors (pypy has some issues with those).